extensions/amp-story-auto-ads/0.1/story-ad-page.js

```diff
@@ -480,24 +502,49 @@ export class StoryAdPage {
       })
     );
 
-    const adChoicesIcon = createElementWithAttributes(
+    this.adChoicesIcon_ = createElementWithAttributes(
       this.doc_,
       'img',
       dict({
         'class': 'i-amphtml-story-ad-attribution',
         'src': src,
       })
     );
+    this.storeService_.subscribe(
+      StateProperty.UI_STATE,
+      uiState => {
+        this.onUIStateUpdate_(uiState);
+      },
+      true /** callToInitialize */
+    );
 
-    adChoicesIcon.addEventListener(
+    this.adChoicesIcon_.addEventListener(
       'click',
       this.handleAttributionClick_.bind(this, href)
     );
 
-    createShadowRootWithStyle(root, adChoicesIcon, attributionCSS);
+    createShadowRootWithStyle(root, this.adChoicesIcon_, attributionCSS);
     this.pageElement_.appendChild(root);
   }
 
+  /**
+   * Reacts to UI state updates and passes the information along as
+   * attributes to the shadowed attribution icon.
+   * @param {!UIType} uiState
+   * @private
+   */
+  onUIStateUpdate_(uiState) {
+    if (!this.adChoicesIcon_) {
+      return;
+    }
+
+    this.adChoicesIcon_.classList.remove(DESKTOP_FULLBLEED_CLASS);
+
+    if (uiState === UIType.DESKTOP_FULLBLEED) {
+      this.adChoicesIcon_.classList.add(DESKTOP_FULLBLEED_CLASS);
+    }
```

Comment:
If you feel like factorizing this in a single line (it's a bit less clear tho, so up to you)
```
this.adChoicesIcon_.classList.toggle(
    DESKTOP_FULLBLEED_CLASS, uiState === UIType.DESKTOP_FULLBLEED);
```